### PR TITLE
Fix help for utf8-strict-mode

### PR DESCRIPTION
**Pull request by grobinson-grafana**
#### What this PR does

This commit fixes the help message for `utf8-strict-mode` as it was out of date and contained incorrect information. The metrics mentioned no longer exist, and the current recommendation is to check the Mimir server logs for information about incompatible tenant configurations.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 1]
```diff
@@ -2435,9 +2435,17 @@ alertmanager_client:
...
 # (experimental) Enable UTF-8 strict mode. Allows UTF-8 characters in the
 # matchers for routes and inhibition rules, in silences, and in the labels for
-# alerts. It is recommended to check both alertmanager_matchers_disagree_total
-# and alertmanager_matchers_incompatible_total metrics before using this mode as
-# otherwise some tenant configurations might fail to load.
+# alerts. It is recommended that all tenants run the `migrate-utf8` command in
+# mimirtool before enabling this mode as otherwise some tenant configurations
+# might fail to load. Tenants with incompatible configurations can be identified
+# in Mimir server logs searching for lines containing `Alertmanager is moving to
+# a new parser for labels and matchers, and this input is incompatible.`. Tenant
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2442]:
```suggestion
# mimirtool before enabling this mode. Otherwise, some tenant configurations
# might fail to load. To identify tenants with incompatible configurations, you can search
# Mimir server logs for lines containing `Alertmanager is moving to
# a new parser for labels and matchers, and this input is incompatible.`.
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 2]
```diff
@@ -2435,9 +2435,17 @@ alertmanager_client:
...
+# configurations that are valid but contain ambiguous matchers can be found
+# searching for log lines containing `Matchers input has disagreement`. Each log
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2444]:
```suggestion
# To find tenant configurations that are valid but contain ambiguous matchers, 
# search for log lines containing `Matchers input has disagreement`. Each log
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 3]
```diff
@@ -2435,9 +2435,17 @@ alertmanager_client:
...
+# line includes the invalid input, a suggestion on how to fix the input, and the
+# ID of the affected tenant. Mimir must be run with debug level logging enabled.
+# More information about this can be found at
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2447]:
```suggestion
# ID of the affected tenant. To use this mode, you must run Mimir with debug-level logging enabled.
# For more information, refer to
```

**Review by tacole02 — APPROVED** [review]
Thank you!

File: docs/sources/mimir/configure/configuration-parameters/index.md (hunk 3 above)
Comment (grobinson-grafana) [reply docs/sources/mimir/configure/configuration-parameters/index.md:2447]: Nit: Debug level logging must be enabled to find affected tenants, not use the mode. Does that change the phrasing at all?

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 4]
```diff
@@ -2435,9 +2435,17 @@ alertmanager_client:
...
+# mimirtool before enabling this mode. Otherwise, some tenant configurations
+# might fail to load. To identify tenants with incompatible configurations,
+# search Mimir server logs for lines containing `Alertmanager is moving to a new
+# parser for labels and matchers, and this input is incompatible.`. To find
+# tenant configurations that are valid but contain ambiguous matchers search for
+# log lines containing `Matchers input has disagreement`. Each log line includes
+# the invalid input, a suggestion on how to fix the input, and the ID of the
+# affected tenant. Mimir must be run with debug level logging enabled. For more
+# information, refer to
+# https://prometheus.io/docs/alerting/latest/configuration/#label-matchers.
```
Comment (tomratcliffe) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2448]: Minor drive-by comment: The link doesn't work because of the fullstop, FYI
```suggestion
# https://prometheus.io/docs/alerting/latest/configuration/#label-matchers
```

Comment (grobinson-grafana) [reply docs/sources/mimir/configure/configuration-parameters/index.md:2449]: Hmm, it works for me?

Comment (grobinson-grafana) [reply docs/sources/mimir/configure/configuration-parameters/index.md:2449]: Oh I see, the jump to `#label-matchers` doesn't work. Will fix! Thanks @tomratcliffe!

Comment (grobinson-grafana) [reply docs/sources/mimir/configure/configuration-parameters/index.md:2449]: Linter might complain though, let's see...

Comment (grobinson-grafana) [reply docs/sources/mimir/configure/configuration-parameters/index.md:2449]: I think this is fine, the other links have a full stop at the end?

File: docs/sources/mimir/configure/configuration-parameters/index.md (hunk 3 above)
Comment (tacole02) [reply docs/sources/mimir/configure/configuration-parameters/index.md]: Yes, thank you for commenting. Updated my suggestion :)

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 5]
```diff
@@ -2435,9 +2435,18 @@ alertmanager_client:
...
-# otherwise some tenant configurations might fail to load.
+# alerts. It is recommended that all tenants run the `migrate-utf8` command in
+# mimirtool before enabling this mode. Otherwise, some tenant configurations
+# might fail to load. To identify tenants with incompatible configurations,
+# search Mimir server logs for lines containing `Alertmanager is moving to a new
+# parser for labels and matchers, and this input is incompatible.`. To find
+# tenant configurations that are valid but contain ambiguous matchers search for
+# log lines containing `Matchers input has disagreement`. Each log line includes
+# the invalid input, a suggestion on how to fix the input (excluding ambiguous
+# matchers as these require manual correction), and the ID of the affected
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2446]:
```suggestion
# matchers, as these require manual correction), and the ID of the affected
```

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 6]
```diff
@@ -2435,9 +2435,18 @@ alertmanager_client:
...
+# tenant. Mimir must be run with debug level logging enabled otherwise these
+# lines will not be logged. For more information, refer to
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:2448]:
```suggestion
# tenant. You must run Mimir with debug-level logging enabled. Otherwise, these
# lines aren't logged. For more information, refer to
```

**Review by tacole02 — APPROVED** [review]

**Review by Logiraptor — APPROVED** [review]
